src/pipx/package_specifier.py

```diff
@@ -188,6 +188,23 @@ def parse_specifier_for_upgrade(package_spec: str) -> str:
     return package_or_url
 
 
+def get_extras(package_spec: str) -> Set[Any]:
+    parsed_package = _parse_specifier(package_spec)
+    if parsed_package.valid_pep508 and parsed_package.valid_pep508.extras is not None:
+        return parsed_package.valid_pep508.extras
+    elif parsed_package.valid_local_path:
+        (_, package_extras_str) = _split_path_extras(parsed_package.valid_local_path)
+        return Requirement("notapackage" + package_extras_str).extras
+
+    return set()
+
+
+def append_extras(package: str, package_or_url: str) -> str:
+    package_spec = Requirement(package)
+    package_spec.extras = get_extras(package_or_url)
+    return str(package_spec)
```

Comment:
The interface around here is pretty weird. What are `package` and `package_or_url` supposed to contain? 

If `package` is only expected to contain the package name, there are better ways to append extras. If it can contain any PEP 508 requirements, this “append” operation does not look right.

Reply:
I'm fine with changing the interface, but basically to do a proper venv inspection for this package we need both the package name, and the extras used to install that package.  We don't need other parts of the package specifier.

Reply:
`package` is just the package name, and `package_or_url` is the full specifier.  This is naming scheme is somewhat legacy for the pipx codebase.

Reply:
> If `package` is only expected to contain the package name, there are better ways to append extras.

Just out of curiosity, what are the better ways you're thinking of?

Reply:
Hmm. I haven’t thought it all through, but since this joined the package name and extras, I would probably make `get_venv_metadata_for_package` and `inspect_venv` take `package: str, extras: Set[str]` instead.

Another ways to do this is to perform marker evaluation. Instead of discarding dependency information with non-matching markers, all dependencies can be stored lossless PEP 508 strings in the metadata storage, and only parsed and filtered when they are actually used. This is cleaner IMO, but may require refactoring too significant to be done with the importlib-metadata migration.

Reply:
I don't totally understand how this would work.  Part of it is that the recursive search includes dependencies of dependencies of ...  And if you include branches of that tree that don't really exist (because they are not needed for you OS, extras, etc.) you are making a lot of work for nothing.

Reply:
As for the interface, I realized after looking again at the code that part of what makes it convenient to have an argument in the format of a requirement, is that that's what the recursive function is expecting.  So I'd probably be combining the name and the extras anyway to supply that function.

Reply:
I changed `get_venv_metadata_for_package` and `inspect_venv take` separate package and extras arguments.